pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -425,6 +437,58 @@ public BrokerResponse handleRequest(JsonNode request, @Nullable RequesterIdentit
     return brokerResponse;
   }
 
+  /**
+   * Set Log2m value for DistinctCountHLL Function
+   * @param brokerRequest
+   * @param hllLog2mOverride
+   */
+  static void handleHyperloglogLog2mOverride(BrokerRequest brokerRequest, int hllLog2mOverride) {
+    if (brokerRequest.getAggregationsInfo() != null) {
+      for (AggregationInfo aggregationInfo : brokerRequest.getAggregationsInfo()) {
+        switch (AggregationFunctionType.valueOf(aggregationInfo.getAggregationType().toUpperCase())) {
+          case DISTINCTCOUNTHLL:
+          case DISTINCTCOUNTHLLMV:
+          case DISTINCTCOUNTRAWHLL:
+          case DISTINCTCOUNTRAWHLLMV:
+            if (aggregationInfo.getExpressionsSize() == 1) {
+              aggregationInfo.addToExpressions(Integer.toString(hllLog2mOverride));
+            }
+        }
+      }
+    }
+    if (brokerRequest.getPinotQuery() != null && brokerRequest.getSelections() != null) {
+      for (Expression expr : brokerRequest.getPinotQuery().getSelectList()) {
+        updateDistinctCountHllExpr(expr, hllLog2mOverride);
+      }
+    }
+  }
+
+  private static void updateDistinctCountHllExpr(Expression expr, int hllLog2mOverride) {
+    if (expr == null || expr.getFunctionCall() == null) {
+      return;
+    }
+    Function functionCall = expr.getFunctionCall();
```

Comment:
`expr` can never be `null` here.
```suggestion
    Function functionCall = expr.getFunctionCall();
    if (functionCall == null) {
      return;
    }
```